src/shapes/src/HollowConicalFrustum.cpp

```diff
@@ -20,55 +20,87 @@
 #include <shapes/HollowConicalFrustum.hpp>
 
 #include <utils/Vector.hpp>
+#include <utils/math/abs.hpp>
 #include <utils/math/coordinate_transformation.hpp>
 
-#include <boost/algorithm/clamp.hpp>
-
-#include <cstdlib>
-
 namespace Shapes {
-
 void HollowConicalFrustum::calculate_dist(const Utils::Vector3d &pos,
                                           double &dist,
                                           Utils::Vector3d &vec) const {
+
   // transform given position to cylindrical coordinates in the reference frame
   // of the cone
-  auto const v = pos - m_center;
+  auto const v = pos - m_cyl_transform_params->center();
   auto const pos_cyl = Utils::transform_coordinate_cartesian_to_cylinder(
-      v, m_axis, m_orientation);
-  // clang-format off
-  /*
-   * the following implementation is based on:
-   *   - defining the cone in the cylindrical 2d coordinates with e_z = m_axis
-   * and r in the plane of axis and pos (1)
-   *   - defining the normal to the cone (2)
-   *   - find the intersection between (1) and (2)
-   *
-   *   r_cone(z) = m * z + (r1 + r2) / 2, -l/2 <= z <= l/2 with m = (r1 - r2) / l
-   *   r_normal(z) = -1/m * z + r_pos + 1 / m * z_pos
-   *   r_cone = r_normal => z_intersection = (-m*(r1+r2-2*r_pos)+2*z_pos) / (2*(1+m**2))
-   */
-  // clang-format on
-  auto const m = (m_r1 - m_r2) / m_length;
-  auto z_intersection = (-m * (m_r1 + m_r2 - 2 * pos_cyl[0]) + 2 * pos_cyl[2]) /
-                        (2 * (1 + m * m));
-  // Limit the possible z values to the range in which the cone exists.
-  z_intersection =
-      boost::algorithm::clamp(z_intersection, -m_length / 2.0, m_length / 2.0);
-  auto r_cone = [this](auto z) {
-    return (m_r1 - m_r2) / m_length * z + (m_r1 + m_r2) / 2.0;
+      v, m_cyl_transform_params->axis(), m_cyl_transform_params->orientation());
+
+  auto project_on_line = [](auto const vec, auto const line_start,
+                            auto const line_director) {
+    return line_start + line_director * ((vec - line_start) * line_director);
   };
-  auto const r_intersection = r_cone(z_intersection);
-  // Transform back to cartesian coordinates.
-  auto const pos_intersection =
-      Utils::transform_coordinate_cylinder_to_cartesian(
-          {r_intersection, pos_cyl[1], z_intersection}, m_axis, m_orientation) +
-      m_center;
 
-  auto const u = (pos - pos_intersection).normalize();
-  auto const d = (pos - pos_intersection).norm() - 0.5 * m_thickness;
-  auto const in_out = (d < 0.0) ? -m_direction : m_direction;
-  dist = in_out * std::abs(d);
+  Utils::Vector3d pos_closest;
```

Comment:
not sure if this is easy to follow... can't you also recycle the old code for the case where the angle is within the segment without the gap?

Reply:
I tried to, but it does not work. The point of interest, ``axis`` and the edge of the mantle are not in a common plane

Reply:
should the distance calculation just the prior on if the point of interest is in the region without the gap? Think of the the case where the central angle is 0 or 2pi (no gap)...

Reply:
We can revert to the old calculation. I thought it is more clear if for both cases the calculation is similar

Reply:
not if you remove all my comments and don't add new ones for your calculation ;). Also you could have saved your time implementing this again... So please add some more comments and explain what you are calculating. Without additional informations it's really hard to follow what you are doing here.